core/src/test/java/cucumber/runtime/ShellwordsTest.java

```
@@ -1,30 +1,189 @@
+                asList("--alfa", "Bravo \" Charlie")
+            },
+            new Object[]{
+                "--delta 'Echo \"foxtrot golf\" hotel'",
+                asList("--delta", "Echo \"foxtrot golf\" hotel")
+            },
+
+            // double_quotes wrapped by single_quote
+            new Object[]{
+                "--india '\"Juliett Kilo\"'",
+                asList("--india", "\"Juliett Kilo\"")
+            },
+
+            // single_quote wrapped by double_quotes
+            new Object[]{
+                "--lima \"'Mike November'\"",
+                asList("--lima", "'Mike November'")
+            },
+
+            // parses_both_single_and_double_quoted_strings
+            new Object[]{
+                "--oscar \"Papa Quebec\" --romeo 'Sierra Tango'",
+                asList("--oscar", "Papa Quebec", "--romeo", "Sierra Tango")
+            },
+
+            // triple double_quote
+            // TODO is this correct with trailing empty string
+            new Object[]{
+                "--uniform \"Victor Whiskey --xray \"Yankee Zulu\"",
+                asList("--uniform", "Victor Whiskey --xray ", "Yankee", "Zulu", "\"")
+            },
+
+            // opening double_quote only with following single_quotes
+            new Object[]{
+                "--alfa \"Bravo Charlie --delta 'Echo Foxtrot'",
+                asList("--alfa", "\"", "Bravo", "Charlie", "--delta", "Echo Foxtrot")
+            },
+
+            // triple single_quotes
+            // TODO is this correct with trailing empty string
+            new Object[]{
+                "--golf 'Hotel India --juliett 'Kilo Lima'",
+                asList("--golf", "Hotel India --juliett ", "Kilo", "Lima", "'")
+            },
+
+            // opening single_quotes only with following double_quotes
+            new Object[]{
+                "--mike 'November Oscar --papa \"Quebec Romeo\"",
+                asList("--mike", "'", "November", "Oscar", "--papa", "Quebec Romeo")
```

This has an unbalanced quote and should throw an error.